src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
+type FigureColor = 'red' | 'green' | 'blue';
```

Comment:
Better to use `enum` here
You will be able to write `shape: FigureShape = FigureShape.triangle;` instead of `shape: FigureShape = 'triangle';`
You will avoid using "magic values"